src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -239,6 +240,67 @@ public ResponseEntity<Exam> updateExam(@PathVariable Long courseId, @RequestBody
+     * @param workingTimeChange the working time change in seconds (can be positive or negative, but must not be 0)
+     * @return the ResponseEntity with status 200 (OK) and with the updated exam as body
+     */
+    @PatchMapping("/courses/{courseId}/exams/{examId}/student-exams/working-time")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Exam> updateExamWorkingTime(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody Integer workingTimeChange) {
+        log.debug("REST request to update the working time of exam : {}", examId);
+
+        examAccessService.checkCourseAndExamAccessForInstructorElseThrow(courseId, examId);
+
+        if (workingTimeChange == 0) {
+            throw new BadRequestException();
+        }
+
+        Exam exam = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(examId);
+        var originalExamDuration = exam.getDuration();
+
+        // 1. Update the end date & working time of the exam
+        exam.setEndDate(exam.getEndDate().plusSeconds(workingTimeChange));
+        exam.setWorkingTime(exam.getWorkingTime() + workingTimeChange);
+        examRepository.save(exam);
+
+        // 2. Re-calculate the working times of all student exams
+        var studentExams = studentExamRepository.findByExamId(examId);
+        for (var studentExam : studentExams) {
+            Integer originalStudentWorkingTime = studentExam.getWorkingTime();
+            int originalTimeExtension = originalStudentWorkingTime - originalExamDuration;
+            // NOTE: take the original working time extensions into account
+            if (originalTimeExtension == 0) {
+                studentExam.setWorkingTime(originalStudentWorkingTime + workingTimeChange);
+            }
+            else {
+                double relativeTimeExtension = (double) originalTimeExtension / (double) originalExamDuration;
+                int adjustedTimeExtension = Math.toIntExact(Math.round(relativeTimeExtension * workingTimeChange));
+                studentExam.setWorkingTime(originalStudentWorkingTime + adjustedTimeExtension);
+            }
+            var savedStudentExam = studentExamRepository.save(studentExam);
+            // NOTE: if the exam already started, notify the student about the working time change
+            if (ZonedDateTime.now().isAfter(exam.getVisibleDate())) {
+                websocketMessagingService.sendMessage(STUDENT_WORKING_TIME_CHANGE_DURING_CONDUCTION_TOPIC.formatted(savedStudentExam.getId()), savedStudentExam.getWorkingTime());
+            }
+        }
+
+        // NOTE: if the exam already started, notify instances about the working time change
+        if (ZonedDateTime.now().isAfter(exam.getVisibleDate())) {
+            instanceMessageSendService.sendExamWorkingTimeChangeDuringConduction(exam.getId());
+        }
+
+        if (ZonedDateTime.now().isBefore(examDateService.getLatestIndividualExamEndDate(exam)) && exam.getStartDate() != null
```

Comment:
Please add a comment here explaining this condition. Why do we use the start date + change?  (If I extend the working time by 2 minutes and I'm 3 minutes into the exam, we don't schedule modeling exercises, but if we are only one minute into the exam we do?

Reply:
I don't know, could also not figure that out, probably @krusche knows what is the rationale behind it?

Reply:
I think I just copied this code from StudentExamResource.updateWorkingTime without thinking too much about it

Reply:
The code was initially written by @iozbeyli as part of #3686
<img width="1293" alt="image" src="https://github.com/ls1intum/Artemis/assets/744067/ff65c06b-79b9-4d68-8a48-8e96a1a6f289">

Reply:
I would assume the idea is that building clusters with compass should only be done after all student exams have finished, so re-scheduling the modeling exercise definitely makes sense, however I agree that the logic is flawed

Reply:
should be fixed now